src/lib.rs

```diff
@@ -92,6 +92,10 @@ pub(crate) fn new_element(name: &str, text: Option<String>) -> Element {
     }
 }
 
+pub trait Build {
+    type Builder;
+}
+
```

Comment:
Why is this not

```suggestion
pub trait Builder {
    type Output;
    type Error: std::error::Error + std::fmt::Display + 'static; // Maybe Send + Sync even?
    pub fn build(self) -> Result<Self::Output, Self::Error>;
}
```
and then move every impl of fn build to the trait. I don't see the reason for having this trait unless it is used like this.

Reply:
I removed this trait as it is not important. We can discuss this in future.